src/datachain/lib/dc.py

```diff
@@ -1263,8 +1263,20 @@ def parse_tabular(
             dc = dc.parse_tabular(format="json")
             ```
         """
+        from pyarrow.dataset import CsvFileFormat, JsonFileFormat
+
         from datachain.lib.arrow import ArrowGenerator, infer_schema, schema_to_output
 
+        if nrows:
+            format = kwargs.get("format")
+            if format not in ["csv", "json"] and not isinstance(
+                format, (CsvFileFormat, JsonFileFormat)
+            ):
+                raise ValueError(
```

Comment:
minor: raise NotImplemented?

Reply:
Changed to `DatasetPrepareError` since that's used as the main type for datachain errors.